lib/spack/spack/database.py

```diff
@@ -1545,9 +1545,14 @@ def _query(
     _query.__doc__ += _query_docstring
 
     def query_local(self, *args, **kwargs):
-        """Query only the local Spack database."""
+        """Query only the local Spack database.
+
+        This function doesn't guarantee any sorting of the returned
+        data for performance reason, since comparing specs for __eq__
```

Comment:
@becker33 This is a typo:
```suggestion
        data for performance reason, since comparing specs for __lt__
```
Sorting uses `<` not, `==`. I'll rebase this PR and ping for a review.